Index: linux-4.4.60/net/sched/sch_hfsc.c
===================================================================
--- linux-4.4.60.orig/net/sched/sch_hfsc.c	2021-03-26 11:22:16.000000000 +0800
+++ linux-4.4.60/net/sched/sch_hfsc.c	2021-07-19 14:36:34.963304779 +0800
@@ -1423,8 +1423,9 @@
 		if (next_time == 0 || next_time > q->root.cl_cfmin)
 			next_time = q->root.cl_cfmin;
 	}
-	WARN_ON(next_time == 0);
-	qdisc_watchdog_schedule(&q->watchdog, next_time);
+
+	if(next_time)
+		qdisc_watchdog_schedule(&q->watchdog, next_time);
 }
 
 static int
