Index: hostapd-2020-10-14/src/ap/acs.c
===================================================================
--- hostapd-2020-10-14.orig/src/ap/acs.c
+++ hostapd-2020-10-14/src/ap/acs.c
@@ -390,7 +390,7 @@ static int acs_usable_bw40_chan(const st
 
 static int acs_usable_bw80_chan(const struct hostapd_channel_data *chan)
 {
-	const int allowed[] = { 5180, 5260, 5550, 5580, 5660, 5680, 5955, 6035, 6115,
+	const int allowed[] = { 5180, 5260, 5550, 5580, 5660, 5680, 5745, 5955, 6035, 6115,
 				6195, 6275, 6355, 6435, 6515, 6595, 6675, 6755, 6835,
 				6915, 6995 };
 	unsigned int i;
